src/spec-configuration/lockfile.ts

```diff
@@ -13,20 +13,8 @@ export interface Lockfile {
 	features: Record<string, { version: string; resolved: string; integrity: string }>;
 }
 
-export async function writeLockfile(params: ContainerFeatureInternalParams, config: DevContainerConfig, featuresConfig: FeaturesConfig, forceInitLockfile?: boolean, dryRun?: boolean): Promise<string | undefined> {
```

Comment:
Strange diff here.  What I did was extract out `generateLockfile` from what was `writeLockfile`.  That eliminates the need to pass in the `dryRun` bool that I added a few commits ago.